src/utils.cpp

```diff
@@ -126,4 +126,94 @@ void get_free_symbolic_consts(const smt::Term &term, smt::TermVec &out)
   }
 }
 
+DisjointSet::DisjointSet(bool (*c)(const smt::Term & a, const smt::Term & b))
+    : comp(c)
+{
+}
+
+DisjointSet::~DisjointSet() {}
+
+void DisjointSet::add(const Term & a, const Term & b)
+{
+  if (leader_.find(a) != leader_.end())
+  {
+    Term leadera = leader_.at(a);
+    UnorderedTermSet & groupa = group_.at(leadera);
+
+    if (leader_.find(b) != leader_.end())
+    {
+      Term leaderb = leader_.at(b);
+
+      if (leadera != leaderb)
+      {
+        UnorderedTermSet & groupb = group_.at(leaderb);
+
+        if (comp(leadera, leaderb))
+        {
+          // Choose according to the given Ranking
```

Comment:
```suggestion
          // Choose according to the given ranking
```
I don't feel that strongly about this btw, just wasn't sure that it needed to be capitalized.